### ISSUE TITLE
rdoc version requirements are possibly too strict

### ISSUE THREAD

**Issue by matt-liu** [SciRuby/nmatrix#309]
We are using nmatrix with a rails 3.2 app.

Our issue is that we cannot use nmatrix without upgrading to rdoc 4.0 and that breaks our dependencies.

Our workaround was to decrease the rdoc requirements to 3.12, and that worked us.

Would you please consider decreasing the requirements in the gemspec to '~> 3.12'?

**Comment by mohawkjohn** [comment]
@matt-liu I'm having trouble figuring out where in Rails the 3.12 requirement is. If I look in `rails.gemspec`, I don't see rdoc at all.

**Comment by mohawkjohn** [comment]
@matt-liu Wanted to ping you again about this. Would you like to submit a pull request?

**Comment by matt-liu** [comment]
Hi John,

The "rdoc 3.12" number was just what was in our Gemfile for our specific project. Sorry for not being more specific about that.

I'm not sure what the best minimum version number of rdoc required would be. I just wanted to let you know that we ended up forking and lowering the requirement for that one reason.

**Comment by mohawkjohn** [comment]
@matt-liu Does it still work if you move rdoc to `add_development_dependency`?

**Comment by matt-liu** [comment]
On a very quick check, it looks like that resolves the issue.

```diff
-  gem.add_dependency 'rdoc', '~>3.12', '>=3.12.2'
+  gem.add_development_dependency 'rdoc', '~>4.0', '>=4.0.1'
```

For what it's worth this was the conflict with rails 3.2.19:
(It was actually 3.4 and not 3.12 we needed)
```
rails (= 3.2.19) ruby depends on
  railties (= 3.2.19) ruby depends on
    rdoc (~> 3.4) ruby
```

**Comment by mohawkjohn** [comment]
Great. Would you be willing to send a pull request with that change? (My Mac's build environment is fubar'd at the moment.)

**Comment by cjfuller** [comment]
Is there a reason that NMatrix needs a `~>` dependency at all?  Presumably this would have worked if we had just had `>= 3.12.2` in the original?

My proposed solution would be: `gem.add_development_dependency 'rdoc', '>=3.12.2'`

**Comment by mohawkjohn** [comment]
Actually, I think there *is* a reason. Most of NMatrix is written in C++, not C, and versions of RDoc earlier than 4.0.1 choked on the C++ method declaration syntax (but not the C version). In other words, if you use RDoc earlier than 4.0.1, it's not going to properly incorporate any of the C-side documentation we've worked so hard on. @agarie, was it you who found that bug?

**Comment by agarie** [comment]
Yes! It's in [rdoc 4.01 History](https://github.com/rdoc/rdoc/blob/master/History.rdoc#401--2013-03-27), "RDoc::Parser::C no longer ignores a (METHOD) cast in rb_define_method.".

**Comment by cjfuller** [comment]
Ok, thanks!  Good to know!

**Comment by mohawkjohn** [comment]
Fixed by #312
